### PR TITLE
Add view of publication

### PR DESCRIPTION
**Pull request by brinxmat**
In this PR:
- Simplifies data for use in graph DB. 
- Restructures data to avoid excessive nodes and removes triples not in queries (removes ca. 75% of the data in a publication)
- I don't like the naming, please suggest something
- This probably needs far more extensive testing, but I suggest we continue and check out a branch for testing

Next steps:
- Wire the code into transformation
- Fix the queries used by the APIs
- Do the same for NVI (I'm assuming that the NVI data has a simpler structure to begin with)

### CODE REVIEW

**Review by ketilaa — APPROVED** [review]
Looks good. Looking forward to see it in use. Sparql is still a bit hard to read to me. No good suggestions for naming.

**Comment by brinxmat** [comment]
> Looks good. Looking forward to see it in use. Sparql is still a bit hard to read to me. No good suggestions for naming.

I updated the code with annotations to make it easier to fathom.

**Review by ketilaa — APPROVED** [review]

File: bulk-load/src/main/java/no/sikt/nva/data/report/api/etl/transformer/ViewCompiler.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    private final Model model;
+
+    public ViewCompiler(InputStream inputStream) {
+        this.model = ModelFactory.createDefaultModel();
+        RDFDataMgr.read(model, inputStream, Lang.JSONLD);
+    }
+
+    public Model extractView() {
+        var query = IoUtils.stringFromResources(Path.of("view_of_publication.sparql"));
```
Comment (truhacevkir) [inline bulk-load/src/main/java/no/sikt/nva/data/report/api/etl/transformer/ViewCompiler.java:22]: Should be constant

File: bulk-load/src/test/java/no/sikt/nva/data/report/api/etl/transformer/ViewCompilerTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class ViewCompilerTest {
+
+    private static final String JOURNAL_JSON = "journal.json";
+    private static final String JOURNAL_NT = "journal.nt";
+
+    @Test
+    void shouldReduceTriplesToViewRequiredToProduceApiData() {
+        var inputStream = IoUtils.inputStreamFromResources(JOURNAL_JSON);
+        var model = new ViewCompiler(inputStream).extractView();
+        assertTrue(expected().isIsomorphicWith(model));
```
Comment (truhacevkir) [inline bulk-load/src/test/java/no/sikt/nva/data/report/api/etl/transformer/ViewCompilerTest.java:20]: new line?

File: bulk-load/src/main/resources/view_of_publication.sparql [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+                ?groupUri :hasPart+ ?affiliationUri .
+              }
+            }
+          }
+        }
+      }
+      FILTER(!ISBLANK(?identity))
+    }
+  }
+}
```
Comment (truhacevkir) [inline bulk-load/src/main/resources/view_of_publication.sparql:229]: new line?

**Review by truhacevkir — CHANGES_REQUESTED** [review]

**Review by truhacevkir — APPROVED** [review]

File: bulk-load/src/main/resources/view_of_publication.sparql [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+# {
+#   "@context": {
+#    ...
+#   },
+#   "id": "https://api.dev.nva.aws.unit.no/publication/018ef6b6d373-a0dac535-b0f4-4116-bef1-6bc153a6ba0b",
+#   "type": "AcademicArticle",
+#   "contributors": [
+#     {
+#       "affiliation": {
+#         "id": "https://example.org/organization/201.15.1.0",
```
Comment (monaullah) [inline bulk-load/src/main/resources/view_of_publication.sparql:20]: The affiliation itself needs to be repeated as the `department` property?

File: bulk-load/src/main/resources/view_of_publication.sparql [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+#         "labels": {
+#           "nb": "Institutt et fag",
+#           "en": "Department of something"
+#         }
+#       },
+#       "identity": {
+#         "id": "https://example.org/person/1"
+#       },
+#       "name": "Marc Tveien",
+#       "role": {
```
Comment (monaullah) [inline bulk-load/src/main/resources/view_of_publication.sparql:36]: Flatten?

File: bulk-load/src/main/resources/view_of_publication.sparql [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+              ?departmentUri :hasPart+ ?affiliationUri .
+              OPTIONAL {
+                ?departmentUri :hasPart ?groupUri .
+                ?groupUri :hasPart+ ?affiliationUri .
+              }
+            }
+          }
+        }
+      }
+      FILTER(!ISBLANK(?identity))
```
Comment (monaullah) [inline bulk-load/src/main/resources/view_of_publication.sparql:226]: Move filter up

File: bulk-load/src/test/resources/journal.json [hunk 7]
```diff
@@ -0,0 +1,241 @@
+{
```
Comment (monaullah) [inline bulk-load/src/test/resources/journal.json:1]: Rename files to AcademicArticle?

File: bulk-load/src/main/java/no/sikt/nva/data/report/api/etl/transformer/ViewCompiler.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
+package no.sikt.nva.data.report.api.etl.transformer;
```
Comment (monaullah) [inline bulk-load/src/main/java/no/sikt/nva/data/report/api/etl/transformer/ViewCompiler.java:1]: Move to commoms?

**Comment by monaullah** [comment]
@brinxmat approved?

**Pull request merged**